File: app/src/main/java/com/example/java_viewmodel/MainActivity.java
package com.example.java_viewmodel;

import androidx.fragment.app.FragmentActivity;

import android.os.Bundle;

import com.example.java_viewmodel.countlist.CountListFragment;
import com.example.java_viewmodel.mainlist.MainFragment;

public class MainActivity extends FragmentActivity implements MainFragment.OnActivityInterActonListener {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        if (savedInstanceState == null){
            getSupportFragmentManager().beginTransaction().replace(R.id.fragment_container, MainFragment.newInstance())
                    .addToBackStack(MainFragment.class.getSimpleName()).commit();
        }
    }

    @Override
    public void onSelectGoToCountList() {
        getSupportFragmentManager().beginTransaction().replace(R.id.fragment_container, CountListFragment.newInstance())
                .addToBackStack(CountListFragment.class.getSimpleName())
                .commit();
    }
}

File: app/src/main/java/com/example/kt_viewmodel/MainActivity.kt
package com.example.kt_viewmodel

import android.os.Bundle
import androidx.fragment.app.FragmentActivity
import com.example.kt_viewmodel.countlist.CountListFragment
import com.example.kt_viewmodel.mainlist.MainFragment
import com.example.kt_viewmodel.mainlist.MainFragment.OnActivityInterActonListener

class MainActivity : FragmentActivity(), OnActivityInterActonListener {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        if (savedInstanceState == null) {
            supportFragmentManager.beginTransaction().replace(R.id.fragment_container, MainFragment.newInstance())
                    .addToBackStack(MainFragment::class.java.simpleName).commit()
        }
    }

    override fun onSelectGoToCountList() {
        supportFragmentManager.beginTransaction().replace(R.id.fragment_container, CountListFragment.newInstance())
                .addToBackStack(CountListFragment::class.java.simpleName)
                .commit()
    }
}